io.c

```diff
@@ -9041,20 +8965,23 @@ rb_io_initialize(int argc, VALUE *argv, VALUE io)
 
 /*
  *  call-seq:
- *     ios.set_encoding_by_bom   -> encoding or nil
+ *    set_encoding_by_bom -> encoding or nil
+ *
+ *  If the stream begins with a BOM, consumes the BOM
```

Comment:
I think we shoud explain what's BOM here, it is not a common knowledge and not exactly easy to Google :)

Reply:
Added link to wikipedia BOM.